compose/config/config.py

```diff
@@ -485,7 +485,6 @@ def resolve_volume_paths(service_dict, working_dir=None):
 
 def resolve_volume_path(volume, working_dir, service_name):
     container_path, host_path = split_path_mapping(volume)
-    container_path = os.path.expanduser(container_path)
```

Comment:
I guess it would expand to the wrong path, that's true.  We added this in 1.4.0, so we should probably call this out in the release notes as a bug fix.